internal/repository/repack.go

```diff
@@ -48,20 +48,33 @@ func Repack(
 	}
 
 	wg, wgCtx := errgroup.WithContext(ctx)
-
 	dstRepo.StartPackUploader(wgCtx, wg)
-	wg.Go(func() error {
-		var err error
-		obsoletePacks, err = repack(wgCtx, repo, dstRepo, packs, keepBlobs, p, logf)
-		return err
-	})
+	obsoletePacks, err = repack(wgCtx, repo, dstRepo, packs, keepBlobs, p, logf)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := wg.Wait(); err != nil {
+	// moved flushing of new packs to here
+	if err := dstRepo.Flush(wgCtx); err != nil {
 		return nil, err
 	}
+
 	return obsoletePacks, nil
 }
 
+// RepackInner is a visible wrapper around repack()
+func RepackInner(ctx context.Context,
```

Comment:
This is a horrible hack, but I get that `Repack` currently doesn't allow implementing this PR. So, let's do it properly instead.

Due to https://github.com/restic/restic/pull/5558 it's no longer possible to separately start the uploader and call flush. This is instead handled automatically via `WithBlobUploader`. 

But I don't see why we couldn't extract the `WithBlobUploader` call out of `Repack`. Then the call has to pass in the `restic.BlobSaver` but that is fine IMHO. I'd then also rename `Repack` to `CopyBlobs`.